Tella/Scenes/HomeView/Views/FileDetailView/FileDetailView.swift

```diff
@@ -30,18 +30,35 @@ struct FileDetailView: View {
         }
         .frame(maxWidth: .infinity, maxHeight: .infinity)
         .background(Styles.Colors.backgroundMain)
-        .fullScreenCover(isPresented: $isEditImagePresent) {
+        .fullScreenCover(isPresented: $isEditFilePresented) {
         } content: {
-            EditImageView(viewModel: EditImageViewModel( mainAppModel: appModel,
-                                                         fileListViewModel: fileListViewModel),
-                          isPresented: $isEditImagePresent)
+            redirectEditFile()
         }
         .environmentObject(fileListViewModel)
         .onAppear(perform: {
             self.fileListViewModel.fileActionSource = .details
         })
     }
     
+    @ViewBuilder
+    func redirectEditFile() -> some View {
+        switch viewModel.currentFile?.tellaFileType {
+        case .image:
+            EditImageView(viewModel: EditImageViewModel( mainAppModel: appModel,
```

Comment:
What do you think about separating this code into two functions called `showEditImageView` and `showEditAudioView`?